bindings/c/foundationdb/CWorkload.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * CWorkload.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#ifndef C_WORKLOAD_H
+#define C_WORKLOAD_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct FDB_future FDBFuture;
+typedef struct FDB_result FDBResult;
+typedef struct FDB_database FDBDatabase;
+typedef struct FDB_transaction FDBTransaction;
+
+typedef struct FDB_promise FDBPromise;
+typedef struct FDB_workload FDBWorkload;
+typedef struct FDB_workloadContext FDBWorkloadContext;
+
+typedef enum FDBSeverity {
```

Comment:
Since translator in `ExternalWorkload.cpp` blindly casts this to FDB C++ Severity, I wonder how we can keep these two struct in sync. Perhaps a unit test?

Reply:
Actually, it's cast to the FDBSeverity in `ClientWorkload.h`. But similar syncing issue, or am I missing something?

Reply:
Yes that's a concern I had too, but did not find a satisfactory solution.
For your second remark about the old `ClientWorkload.h`, maybe I misunderstood your problem or the code, but I don't think there was  a syncing issue, the `FDBSeverity` type used in the `ExternalWorkload` is the one defined in `ClientWorkload.h`, it is imported here:
https://github.com/apple/foundationdb/blob/85f293fe58ca270885055d4f2cb4ea0bc42e226f/fdbserver/workloads/ExternalWorkload.actor.cpp#L24
and it is explicitly converted to the internal `Severity` type here:
https://github.com/apple/foundationdb/blob/85f293fe58ca270885055d4f2cb4ea0bc42e226f/fdbserver/workloads/ExternalWorkload.actor.cpp#L62-L83

Maybe to solve this issue the `FDBSeverity` from `CppWorkload.h` should be derived from the C one. I can see two ways of doing it: creating a "CommonWorkload.h` which is included in both the C and C++ bindings, or directly including the C bindings in the C++ ones. I like having a single header file, but I am sure it also has its limitations.
```c++
// CppWorkload.h
namespace capi {
#include "CWorkload.h"
}

enum class FDBSeverity {
    Debug = capi::FDBSeverity_Debug
    Info = capi::FDBSeverity_Info
    Warn = capi::FDBSeverity_Warn
    WarnAlways = capi::FDBSeverity_WarnAlways
    Error = capi::FDBSeverity_Error
}
```